### ISSUE TITLE
Incorrect hook code generation

### ISSUE THREAD

**Issue by RenaKunisaki** [satisfactorymodding/SatisfactoryModLoader#231]
When generating a trampoline to return from a hook function, the `jmp` instruction is used (0xE9 0xCD). This fails, causing a crash, if the jump destination is more than 2GB away.

Reproducing the actual crash is somewhat involved (I have a reliable method, but it depends heavily on my specific system and config) and I don't know exactly which function generates the faulty hook, but the result in my case is:

1. Ficsit-Networks hooks `FactoryGame-Engine-Win64-Shipping!?EndPlay@USceneComponent@@UEAAXW4Type@EEndPlayReason@@@Z`
2. A `jmp` instruction is inserted into that function, jumping to FIN's code
3. FIN's code returns by jumping to a trampoline which reproduces the instructions that the `jmp` overwrote, then jumps back to just after the `jmp`
4. The trampoline looks like:

```
0x91e61020:	rex push %rbp
0x91e61022:	push   %rbx
0x91e61023:	push   %rdi
0x91e61024:	lea    -0x70(%rsp),%rbp
0x91e61029:	jmp    0x101cf5be9
```

The intended destination address is `0x01cf5be9`, but since this is more than 2GB from the trampoline, the jump resolves to `0x101cf5be9` instead. (The actual addresses will vary)

Once I have the game loaded in gdb and the trampoline has been created, I can fix the bug by writing at 0x91e61029: 
```
48 c7 c0 e9 5b cf 01            mov rax, 0x01cf5be9
ff e0                           jmp rax
```

This bug might only manifest on Linux, and depends on having a certain number of mods loaded.

**Comment by mircearoata** [comment]
We are using [kubo/funchook](https://github.com/kubo/funchook) for the function hooking, which should be already checking that the distance is jmp-able, though admittedly a very old version ([this commit](https://github.com/kubo/funchook/tree/dade08843bc18a3aff99f03e0f183aef71070cc8) as far as I could tell from the timestamp of the SML commit adding it).

I already wanted to update to the latest version for a while now, hoping to solve the `Failed to execute VirtualQuery` issue, though that turned out to be happening because there was no page large enough available after the function, and funchook on Windows doesn't check the pages before the function, and it looks like it doesn't do that in the latest version either.

If you are playing on Update 8 (SML >=3.5.0), can you set `enableFunchookLogging` to `true` in `<gameDir>/FactoryGame/Config/SML.cfg`, and send the `SML_funchook.log` from the same folder as the game logs?

**Comment by RenaKunisaki** [comment]
[SML_funchook.log](https://github.com/satisfactorymodding/SatisfactoryModLoader/files/12777237/SML_funchook.log)

It's about 10MB, but there you go!

**Comment by budak7273** [comment]
Should™ be resolved in v3.7.0
